### ISSUE TITLE
getRawValue doesn't look at the raw value

### ISSUE THREAD

**Issue by shawkins** [smallrye/smallrye-config#1053]
Is this correct - https://github.com/smallrye/smallrye-config/blob/5a96de2849c87d144586df07dd4529e28d7e3b85/implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java#L370

It seems like it should consult the raw value, or have some javadocs redefining what raw means for this method.

**Comment by radcortez** [comment]
Yes, it is correct. The names are a bit unfortunate. `getRawValue` is part of the API since the early versions and the intent is to provide the value of the configuration before applying conversions.

Expressions and `ConfigValue` came later, requiring a way to access the value without expansion, which also got the name `getRawValue`.

**Comment by radcortez** [comment]
`SmallRyeConfig#getRawValue` has been deprecated and will be removed in the future.
